cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -21,52 +21,47 @@ import (
 	"net/http"
 
 	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
-	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/cloudformation"
-	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
-	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
-	"github.com/spf13/cast"
-	"go.mongodb.org/atlas/mongodbatlas"
+	"go.mongodb.org/atlas-sdk/v20231001001/admin"
 )
 
+func setup() {
+	util.SetupLogger("mongodb-atlas-global-cluster-config")
+}
+
 var RequiredFields = []string{constants.ClusterName, constants.ProjectID}
 
 func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	setup() // logger setup
+	setup()
 
-	// Validate required fields in the request
-	if modelValidation := validateModel(RequiredFields, currentModel); modelValidation != nil {
-		return *modelValidation, nil
+	if errValidation := validateModel(RequiredFields, currentModel); errValidation != nil {
+		return *errValidation, nil
 	}
 
-	// Create MongoDb Atlas Client using keys
-	if currentModel.Profile == nil || *currentModel.Profile == "" {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
-	}
-	client, pe := util.NewMongoDBClient(req, currentModel.Profile)
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	client, pe := util.NewAtlasClient(&req, currentModel.Profile)
 	if pe != nil {
-		_, _ = logger.Warnf("CreateMongoDBClient error: %v", *pe)
```

Comment:
some logger may not be needed

Reply:
this log seems needed since we do not return an error in this case

Reply:
I think we do return an error in progress event, this code is the same in all resources and we don't log it

Reply:
sorry, I meant the log at line 240. This is why the logger import is still needed